tests/part_test.py

```diff
@@ -912,6 +912,69 @@ def test_get_part_by_pos(self):
         self.assertEqual(ret.start, ps4.start + ps4.size)
         self.assertLessEqual(ret.size, (100 * 1024**2) - (ps4.start + ps4.size))
 
+class PartCreateResizePartCase(PartTestCase):
+    def test_create_resize_part_simple(self):
+        """Verify that it is possible to create and resize a parition"""
+
+        # we first need a partition table
+        succ = BlockDev.part_create_table (self.loop_dev, BlockDev.PartTableType.MSDOS, True)
+        self.assertTrue(succ)
+
+        # create a maximal partition
+        ps = BlockDev.part_create_part (self.loop_dev, BlockDev.PartTypeReq.NORMAL, 2048*512, 0, BlockDev.PartAlign.OPTIMAL)
+        initial_start = ps.start
+        initial_size = ps.size
+
+        new_size = 20 * 1000**2  # resize to 20 MB
+        succ = BlockDev.part_resize_part (self.loop_dev, ps.path, new_size, BlockDev.PartAlign.OPTIMAL)
+        self.assertTrue(succ)
+        ps = BlockDev.part_get_part_spec(self.loop_dev, ps.path)
+        self.assertEqual(initial_start, ps.start)  # offset must not be moved
+        self.assertGreaterEqual(ps.size, new_size)  # at least the requested size,
+        self.assertLess(ps.size, new_size + 8 * 1024**2)  # but also not too big
+
+        # resize to maximum
+        succ = BlockDev.part_resize_part (self.loop_dev, ps.path, 0, BlockDev.PartAlign.OPTIMAL)
+        self.assertTrue(succ)
+        ps = BlockDev.part_get_part_spec(self.loop_dev, ps.path)
+        self.assertEqual(initial_start, ps.start)
+        self.assertEqual(initial_size, ps.size)  # should grow to the same size again
+
+        # resize to maximum explicitly
+        succ = BlockDev.part_resize_part (self.loop_dev, ps.path, initial_size, BlockDev.PartAlign.OPTIMAL)
+        self.assertTrue(succ)
+        ps = BlockDev.part_get_part_spec(self.loop_dev, ps.path)
+        self.assertEqual(initial_start, ps.start)
+        self.assertGreaterEqual(ps.size, initial_size) # at least the requested size
+
+        new_size = 20 * 1000**2  # resize to 20 MB with no alignment
+        succ = BlockDev.part_resize_part (self.loop_dev, ps.path, new_size, BlockDev.PartAlign.NONE)
+        self.assertTrue(succ)
+        ps = BlockDev.part_get_part_spec(self.loop_dev, ps.path)
+        self.assertEqual(initial_start, ps.start)  # offset must not be moved
+        self.assertGreaterEqual(ps.size, new_size)  # at least the requested size,
+        self.assertLess(ps.size, new_size + 8 * 1024**2)  # but also not too big
+
+        # resize to maximum with no alignment
+        succ = BlockDev.part_resize_part (self.loop_dev, ps.path, 0, BlockDev.PartAlign.NONE)
+        self.assertTrue(succ)
+        ps = BlockDev.part_get_part_spec(self.loop_dev, ps.path)
+        new_size = ps.size
+        self.assertEqual(initial_start, ps.start)
```

Comment:
The problem is in libparted which can give a smaller size for maximizing a partition with no alignment than with optimal.